packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3419,17 +3419,19 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
   /// until it finds a [RenderObjectElement].
   RenderObject? get renderObject {
     RenderObject? result;
-    void visit(Element element) {
-      assert(result == null); // this verifies that there's only one child
-      if (element._lifecycleState == _ElementLifecycle.defunct) {
-        return;
-      } else if (element is RenderObjectElement) {
-        result = element.renderObject;
+    final Queue<Element> elements = Queue<Element>();
+    elements.addFirst(this);
+    while (elements.isNotEmpty) {
+      final Element current = elements.removeFirst();
+      if (current._lifecycleState == _ElementLifecycle.defunct) {
+        break;
+      } else if (current is RenderObjectElement) {
+        result = current.renderObject;
```

Comment:
FYI there is a slight change here that may not be a problem.  The old version, if you have multiple children that are RenderObjectElement, the last one visited will be returned.  This implementation stops execution at the first one found.

Reply:
I'm surprised by the old behavior. I wonder if that was intentional or not?

Reply:
It's hard for me to believe it was intentional, the documentation states that the order of `visitChildren` is undefined so it can't be meaningful that the last one get precedence.

Reply:
Is there really a functional change here? The old code contained this assert: `assert(result == null);` which was enforcing that there can only ever be one child. If there's only ever one child, then it doesn't matter whether you return the first one or last one. It's the same one?

Reply:
In fact, we should keep that assert around, adapted to the new code. It is invalid to call this if the Element has multiple children. An assert should warn you about this instead of just silently returning a "random" child.

Reply:
Also, since there's only going to be one child, you probably don't even need the `Queue`?

Reply:
@goderbauer good call, I removed the assert when I was writing the benchmark to make sure it didn't mess up the results and missed it's purpose.  I made the change, it is surprisingly only slightly faster than the  one that allocates the queue.